pkg/channel/jetstream/dispatcher/consumer.go

```diff
@@ -144,61 +248,76 @@ func (c *Consumer) doHandle(ctx context.Context, msg *nats.Msg) protocol.Result
 		logger.Debugw("received message from JetStream consumer", debugKVs...)
 	}
 
-	message := cejs.NewMessage(msg)
-	if message.ReadEncoding() == binding.EncodingUnknown {
+	if msg.ReadEncoding() == binding.EncodingUnknown {
 		return errors.New("received a message with unknown encoding")
 	}
 
-	event := tracing.ConvertNatsMsgToEvent(c.logger.Desugar(), msg)
-	additionalHeaders := tracing.ConvertEventToHttpHeader(event)
+	attempt := 1
 
-	sc, ok := tracing.ParseSpanContext(event)
-	var span *trace.Span
-	if !ok {
-		c.logger.Warn("Cannot parse the spancontext, creating a new span")
-		c.ctx, span = trace.StartSpan(c.ctx, jsmChannel+"-"+string(c.sub.UID))
-	} else {
-		c.ctx, span = trace.StartSpanWithRemoteParent(c.ctx, jsmChannel+"-"+string(c.sub.UID), sc)
+	meta, err := msg.Metadata()
+	if err == nil {
+		attempt = int(meta.NumDelivered)
 	}
 
-	defer span.End()
-
 	te := TypeExtractorTransformer("")
 
 	dispatchExecutionInfo, err := c.dispatcher.SendMessage(
```

Comment:
The main problem here is:
- you do not know is it failed to send to Subscriber or to ReplyUrl, in case it failed to send to ReplyUrl it will nack msg, right?
- if you pass noRetry then it is used to for all targets Subscriber, DLQ, ReplyURL; we only want noRetry for send to target

Reply:
Ahhhh good point! I had zero thought on the `ReplyURL`, this does change things considerably.